helm/aws-load-balancer-controller/templates/deployment.yaml

```diff
@@ -27,13 +27,12 @@ spec:
         {{- toYaml .Values.podLabels | nindent 8 }}
         {{- end }}
       annotations:
-        {{- if not .Values.serviceMonitor.enabled }}
```

Comment:
Why so complicated?
IMO using something like I already suggested in comment https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3597#discussion_r1519577932 would be perfectly fine:

> Wrong usage of helm `and` keyword.
> Correct is
> `{{- if and (not .Values.serviceMonitor.enabled) .Values.prometheusScrape.enabled }}`